paddle/fluid/operators/gather_op.cu

```diff
@@ -41,6 +41,7 @@ class GatherOpCUDAKernel : public framework::OpKernel<T> {
         paddle::framework::DataTypeToString(index_type),
         paddle::framework::DataTypeToString(framework::proto::VarType::INT32),
         paddle::framework::DataTypeToString(framework::proto::VarType::INT64));
+    PADDLE_ENFORCE_GT(index->numel(), 0, "The index is empty.");
```

Comment:
The error msg is too simple. The index of gather_op should not be empty?

Reply:
Done